weave/trace/api.py

```diff
@@ -120,6 +120,23 @@ def publish(obj: Any, name: str | None = None) -> weave_client.ObjectRef:
     Returns:
         A weave Ref to the saved object.
     """
+    import weave
+
+    if not isinstance(obj, weave.Object):
+        return _publish(obj, name)
+
+    old_ref, obj.ref = obj.ref, None
```

Comment:
can you elaborate on why this is here?

Reply:
Added a comment to this part of the code